### ISSUE TITLE
RWD shows up as SR+ despite SR+ not being sold for 2.5 years

### ISSUE THREAD

**Issue by yigitgungor** [teslamate-org/teslamate#3887]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Any base model Model 3 shows up as SR+ despite different battery sizes, model years, and feature differences.

### Expected Behavior

2022 and 2023 base model Model 3s should show up as RWD instead of SR+

### Steps To Reproduce

Home Page
Car Model Name

### Relevant log output

```shell
N/A
```

### Type of installation

Docker

### Version

1.28.5

**Comment by yigitgungor** [comment]
I believe this is the mapping for the model names, but I am not sure how to specify RWD cars.

https://github.com/teslamate-org/teslamate/blob/a15631f2ff18b99023d562897de0f56ee3c38c0b/lib/teslamate/vehicles/vehicle.ex#L72

**Comment by famaia** [comment]
I can confirm this with my M3 RWD standard 2024 highland

**Comment by famaia** [comment]
I can confirm this with my M3 RWD standard 2024 highland

**Comment by swiffer** [comment]
what is returned by the vehicle_config for model3 rwd?

https://github.com/teslamate-org/teslamate/issues/2494

**Comment by yigitgungor** [comment]
is there a shell command I can run to dump the result of the vehicle_config call?

**Comment by tomassanjuann** [comment]
> what is returned by the vehicle_config for model3 rwd?
> 
> #2494

`        "vehicle_config": {
            "aux_park_lamps": "None",
            "badge_version": 1,
            "can_accept_navigation_requests": true,
            "can_actuate_trunks": true,
            "car_special_type": "base",
            "car_type": "model3",
            "charge_port_type": "CCS",
            "cop_user_set_temp_supported": false,
            "dashcam_clip_save_supported": true,
            "default_charge_to_max": true,
            "driver_assist": "TeslaAP4",
            "ece_restrictions": true,
            "efficiency_package": "M3POPPYSEED2024",
            "eu_vehicle": true,
            "exterior_color": "UltraRed",
            "exterior_trim": "Black",
            "exterior_trim_override": "",
            "has_air_suspension": false,
            "has_ludicrous_mode": false,
            "has_seat_cooling": true,
            "headlamp_type": "Global",
            "interior_trim_type": "Black2",
            "key_version": 2,
            "motorized_charge_port": true,
            "paint_color_override": "",
            "performance_package": "Base",
            "plg": true,
            "pws": true,
            "rear_drive_unit": "PM216MOSFET",
            "rear_seat_heaters": 1,
            "rear_seat_type": 0,
            "rhd": false,
            "roof_color": "RoofColorGlass",
            "seat_type": null,
            "sentry_preview_supported": false,
            "spoiler_type": "None",
            "sun_roof_installed": null,
            "supports_qr_pairing": false,
            "third_row_seats": "None",
            "timestamp": 1720772906758,
            "trim_badging": "50",
            "use_range_badging": true,
            "utc_offset": 7200,
            "webcam_selfie_supported": true,
            "webcam_supported": true,
            "wheel_type": "Glider18"
        },`

**Comment by tomassanjuann** [comment]
Anything more needed? I can provide whatever needed to solve this. It's strange to see my M3 Highland as SR+ 😂

**Comment by swiffer** [comment]
🤣 haven't found the time to look into it yet. Will try this week

**Comment by yigitgungor** [comment]
Yeah, so looks like the problem is they still use "trim_badging": "50" which is indistinguishable for  SR+ cars and RWD cars.

Probably "efficiency_package": "M3POPPYSEED2024" would be the best one to use to identify.

@tomassanjuann what did you run to get this result, so I can run it for a 2023 RWD too and we have both values to filter on.

**Comment by tomassanjuann** [comment]
> Yeah, so looks like the problem is they still use "trim_badging": "50" which is indistinguishable for SR+ cars and RWD cars.
> 
> Probably "efficiency_package": "M3POPPYSEED2024" would be the best one to use to identify.
> 
> @tomassanjuann what did you run to get this result, so I can run it for a 2023 RWD too and we have both values to filter on.

Make a GET request to /api/1/vehicles/{vehicle_tag}/vehicle_data

https://developer.tesla.com/docs/fleet-api?shell#vehicle_data

**Comment by klavo** [comment]
The problem is that Tesla reused `trim_badging` code "50" for both the discontinued Standard Range Plus (sold through ~Q1 2021) and the current RWD trim, so a static string lookup can't tell them apart — it needs to also account for the car's model year.

Added a `model_year/1` helper to `TeslaMate.Log.Car` that decodes the model year from VIN position 10 (standard SAE J853 code), then used it to disambiguate in `identify/1`:

**`lib/teslamate/log/car.ex`**

```elixir
# SAE J853 VIN position-10 model-year code, valid for the 2010-2039 cycle
# (Tesla has no vehicles from the prior 1980-2009 cycle).
@vin_year_codes %{
  "A" => 2010, "B" => 2011, "C" => 2012, "D" => 2013, "E" => 2014,
  "F" => 2015, "G" => 2016, "H" => 2017, "J" => 2018, "K" => 2019,
  "L" => 2020, "M" => 2021, "N" => 2022, "P" => 2023, "R" => 2024,
  "S" => 2025, "T" => 2026, "V" => 2027, "W" => 2028, "X" => 2029,
  "Y" => 2030, "1" => 2031, "2" => 2032, "3" => 2033, "4" => 2034,
  "5" => 2035, "6" => 2036, "7" => 2037, "8" => 2038, "9" => 2039
}

def model_year(%__MODULE__{vin: vin}), do: model_year(vin)

def model_year(vin) when is_binary(vin) and byte_size(vin) >= 10 do
  Map.get(@vin_year_codes, String.at(vin, 9))
end

def model_year(_vin), do: nil
```

**`lib/teslamate/vehicles/vehicle.ex`**

```diff
-  def identify(%Vehicle{display_name: name, vehicle_config: config}) do
+  def identify(%Vehicle{display_name: name, vehicle_config: config, vin: vin}) do
     case config do
       %VehicleConfig{
         car_type: type,
@@ -87,6 +87,8 @@
             end
           end

+        model_year = Car.model_year(vin)
+
         marketing_name =
           case {model, trim_badging, type} do
             {"S", "100D", "lychee"} -> "LR"
@@ -96,7 +98,10 @@
             {"3", "74D", _} -> "LR AWD"
             {"3", "74", _} -> "LR"
             {"3", "62", _} -> "MR"
-            {"3", "50", _} -> "SR+"
+            # Tesla reused code "50" for both the discontinued Standard Range
+            # Plus (sold through ~Q1 2021) and the current RWD trim.
+            {"3", "50", _} when is_integer(model_year) and model_year < 2021 -> "SR+"
+            {"3", "50", _} -> "RWD"
             {"X", "100D", "tamarind"} -> "LR"
             {"X", "P100D", "tamarind"} -> "Plaid"
             {"Y", "P74D", _} -> "LR AWD Performance"
```

As a bonus, since `model_year` is now available, I also surfaced it in the car summary UI so the model year shows alongside the trim, e.g. "2022 Model 3 RWD":

**`lib/teslamate_web/live/car_live/summary.html.heex`**

```diff
         <%= unless is_nil(@car.model) do %>
           <p class="subtitle is-6 has-text-weight-light">
+            <%= if year = TeslaMate.Log.Car.model_year(@car) do %>
+              <%= year %>
+            <% end %>
             Model <%= @car.model %>
             <%= if @car.marketing_name != nil do %>
               <span
```
